client/human/HumanClientApp.cpp

```diff
@@ -1238,3 +1243,42 @@ void HumanClientApp::OpenURL(const std::string& url) {
     // execute open command
     system(command.c_str());
 }
+
+void HumanClientApp::AddWaitingFlag(std::string reason) {
+    if (reason.empty()) {
+        ErrorLogger() << "Requested addition of empty wait flag";
+        return;
+    }
+
+    TraceLogger() << "Request to add wait flag: " << reason;
+    m_wait_flags[reason]++;
```

Comment:
is this thread safe?
if any networking events can cause this to be called, it might need to be...